service/src/main/java/tds/config/services/impl/TimeLimitConfigurationServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package tds.config.services.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import tds.config.TimeLimitConfiguration;
+import tds.config.repositories.TimeLimitConfigurationRepository;
+import tds.config.services.TimeLimitConfigurationService;
+
+import java.util.Optional;
+
+@Service
+public class TimeLimitConfigurationServiceImpl implements TimeLimitConfigurationService {
+    private final TimeLimitConfigurationRepository timeLimitConfigurationRepository;
+
+    @Autowired
+    public TimeLimitConfigurationServiceImpl(TimeLimitConfigurationRepository timeLimitConfigurationRepository) {
+        this.timeLimitConfigurationRepository = timeLimitConfigurationRepository;
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(String clientName) {
+        return timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName);
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(String clientName, String assessmentId) {
+        return timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName, assessmentId)
+                .map(Optional::of)
```

Comment:
So I looked into this a bit more, and I'm not 100% sure that `orElseGet()` is the right thing to do here.  The intent of `orElseGet()` is to always return _something_ (e.g. a default version of whatever you're trying to get).  So I think `orElseGet()` here is expecting to bring back a `TimeLimitConfiguration` (well, technically a `Supplier<TimeLimitConfiguration>`, not another `Optional<TimeLimitConfiguration>` (which in our case could _still_ be null/return zero records).  So I need to map my result to an `Optional<TimeLimitConfiguration>`, even though under the covers it really _is_ an `Optional<TimeLimitConfiguration>`.

My understanding of the intent of `orElseGet()` is to always return _something_, and in this case we're returning something else that might also be nothing.  So... I dunno.  This might be too "fancy" for its own good; maybe an old-fashioned `if...else` statement will do here.